### PR TITLE
Add support for user information in templates

### PR DESCRIPTION
**Pull request by acmeguy**
Hi,

Here is a minor change that allows users to information about the current user in queries.
Hope it is well received.

Regards,
 - Stefán

### CODE REVIEW

**Comment by acmeguy** [comment]
Added the ability to select natural breakdown of information based on the length of the period being visualized

**Comment by hagen1778** [comment]
Hi @acmeguy !

Sorry for the very late response.

Could you plz provide a few examples where user information and `$naturalTimeSeries` will be useful?

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
+dist/*
```
Comment (Slach) [inline .gitignore:1]: unfortunately `dist` is required folder for submit grafana plugin to grafana-plugins-repository

File: src/sql_query.ts [hunk 2]
```diff
@@ -371,6 +380,35 @@ export default class SqlQuery {
...
+        var FEW_MONTHS = 60 * 60 * 24 * 30 * 10;
+        var FEW_YEARS = 60 * 60 * 24 * 365 * 6;
+        if (dateTimeType === 'DATETIME') {
+            var duration = to - from;
+
+            if (duration < SOME_MINUTES) return 'toUInt32(timestamp) * 1000';
+            else if (duration < FEW_HOURS) return 'toUInt32(toStartOfMinute(timestamp)) * 1000';
+            else if (duration < SOME_HOURS) return 'toUInt32(toStartOfFiveMinute(timestamp)) * 1000';
+            else if (duration < MANY_HOURS) return 'toUInt32(toStartOfFifteenMinutes(timestamp)) * 1000';
+            else if (duration < FEW_DAYS) return 'toUInt32(toStartOfHour(timestamp)) * 1000';
```
Comment (Slach) [inline src/sql_query.ts:400]: look like  we shall use $dateTimeCol here instead of `timestamp`

**Comment by Slach** [comment]
Dear @acmeguy, could you resolve conflicts, remove .dist from .gitignore, and provide jest unit tests that covered your and try to test your changes with grafana7 docker-compose environment?

Also we need improve README.md where explain when and why we need use $naturalTimeSeries

Unfortunately, we can't merge significant changes in query macros without coverage by a unit test

**Comment by Slach** [comment]
@acmeguy sorry for send you message again ;) maybe you can add permission for @Slach account in your repository and I could resolve conflicts for your branch?

**Comment by acmeguy** [comment]
I will gladly attend to this soon  I had not noticed earlier requests

**Comment by Slach** [comment]
Hello @acmeguy any news about this pull request?

**Comment by Slach** [comment]
@acmeguy any news about this PR?

**Comment by Slach** [comment]
@acmeguy according to
https://github.com/grafana/grafana/issues/4922
latest version of grafana succefully support embedded templating variables

```
${__user.email}
${__user.name}
${__user.login}
${__org.name}
```

**Pull request closed without merging**